src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +121,29 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        internal static TAttribute GetResolvedAttribute<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            var attribute = parameter.GetCustomAttribute<TAttribute>();
+
+            var attributeConnectionProvider = attribute as IConnectionProvider;
+            if (attributeConnectionProvider != null && string.IsNullOrEmpty(attributeConnectionProvider.Connection))
+            {
+                // if the attribute doesn't specify an explicit connnection, walk up
+                // the hierarchy looking for an override specified via attribute
+                var connectionProviderAttribute = attribute.GetType().GetCustomAttribute<ConnectionProviderAttribute>();
```

Comment:
ConnectionProviderAttribute is the way the various attributes indicate which override attribute they are bound to. E.g. Queue/Blob/Table attributes are marked with [ConnectionProviderAttribute(typeof(StorageAccountAttribute))]. This is the key to enabling this code to generalize and handle both SAA/SBAA.